KarhooUISDK/Screens/QuoteList/QuoteListFilters/FiltersViews/FilterViewBuilder.swift

```diff
@@ -32,12 +55,61 @@ struct FilterViewBuilder {
         let filter = (filters.first { $0.filterCategory == .luggage } as? QuoteListNumericFilter) ?? QuoteListFilters.LuggageCapacityModel(value: 0)
         return NumericFilterView(filter: filter)
     }
+
+    private func buildVehicleTypeFilterView() -> UIView {
```

Comment:
instead of having 5 methods with the same internal structure, couldn't we have one method that takes the `filterCategory` and `selectableFilters` as parameters and call that 5 times?

Reply:
I tried it already, and it doesn't change a lot of, considering the length of the file. We'd need to pass 3 parameters: selected filters, selectable filters and category. Therefore, the method call length is almost the same as creating the actual specialized method. Additionally, plus of current approach is clear naming and (fairly) easy-to-follow construction

Reply:
Creating helper method with structure/3 parameters has limited outcome. Instead of that, we could just use proper view constructor inside the array. But still in both those approaches we'd take brevity over clarity, which is not good imho. From SRP point of view using several separated methods we provide isolation. The array does not care how a given view is created and what exactly is needed to do this. I know it could be faster, but faster is not always better, considering maintenance.

Reply:
Given it's only 5 small methods, we can leave it like this.